src/data/graphql/typeDefs.ts

```diff
@@ -428,6 +434,7 @@ export default gql`
     nativeTokenAddress: String!
     tokenAddresses: [String]!
     extensionAddresses: [String!]
+    safes: [ColonySafe!]!
```

Comment:
Think this should be `safes: [ColonySafe]!` because the array may be empty?

Incidentally, I made the same changes to the query/resolver [here](https://github.com/JoinColony/colonyDapp/pull/3660). Just flagging for your attention.

Reply:
`ColonySafe!` is not telling GraphQL that the array cannot be empty, it just telling it that the values inside the array are non-nullable so it shouldn't return something like `[null, {...NORMAL_ITEM_HERE}]`, for example. 

`[ColonySafe!]!` is basically telling GraphQL that we are expecting an array, empty or not, that stores `ColonySafe` objects inside it (no nulls).

More info here in the docs: https://graphql.org/learn/schema/

Reply:
I've looked over that before but was confused by "And since Episode! is also non-nullable, you can always expect every item of the array to be an Episode object." But I guess it means _when_ the array contains more than zero items , not _that_ it must return at least one item that is an Episode (or ColonySafe in this case). 

Thanks for clarifying.

Reply:
No problem! At first the use of ! for cases like this can be confusing